pennylane/compiler/qjit_api.py

```diff
@@ -679,17 +684,21 @@ def _call_capture_enabled(self, *init_state):
 
         for_loop_prim = _get_for_loop_qfunc_prim()
 
-        jaxpr_body_fn = jax.make_jaxpr(self.body_fn)(0, *init_state)
+        flat_fn = FlatFn(self.body_fn)
```

Comment:
What prevents us from storing `flat_fn` as `self.flat_fn` instead of wrapping `body_fn` with each call?